bundles/org.openhab.binding.growatt/src/main/java/org/openhab/binding/growatt/internal/action/GrowattActions.java

```diff
@@ -0,0 +1,92 @@
+import org.openhab.core.automation.annotation.ActionInput;
+import org.openhab.core.automation.annotation.RuleAction;
+import org.openhab.core.thing.binding.ThingActions;
+import org.openhab.core.thing.binding.ThingActionsScope;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Implementation of the {@link ThingActions} interface used for setting up battery charging and discharging programs.
+ *
+ * @author Andrew Fiddian-Green - Initial contribution
+ */
+@ThingActionsScope(name = "growatt")
+@NonNullByDefault
+public class GrowattActions implements ThingActions {
+
+    private final Logger logger = LoggerFactory.getLogger(GrowattActions.class);
+    private @Nullable GrowattInverterHandler handler;
+
+    public static void setupChargingProgram(ThingActions actions, Number chargingPower, Number targetSOC,
+            boolean allowAcCharging, String startTime, String stopTime, boolean programEnable) {
+        ((GrowattActions) actions).setupChargingProgram(chargingPower, targetSOC, allowAcCharging, startTime, stopTime,
+                programEnable);
+    }
+
+    public static void setupDischargingProgram(ThingActions actions, Number dischargingPower, Number targetSOC,
+            String startTime, String stopTime, boolean programEnable) {
+        ((GrowattActions) actions).setupDischargingProgram(dischargingPower, targetSOC, startTime, stopTime,
+                programEnable);
+    }
+
+    @Override
+    public @Nullable ThingHandler getThingHandler() {
+        return handler;
+    }
+
+    @Override
+    public void setThingHandler(@Nullable ThingHandler handler) {
+        this.handler = (handler instanceof GrowattInverterHandler growattHandler) ? growattHandler : null;
+    }
+
+    @RuleAction(label = "@text/actions.charging.label", description = "@text/actions.charging.description")
+    public void setupChargingProgram(
+            @ActionInput(name = "charging-power", label = "@text/actions.charging-power.label", description = "@text/actions.charging-power.description") Number chargingPower,
+            @ActionInput(name = "target-soc", label = "@text/actions.target-soc.label", description = "@text/actions.target-soc.description") Number targetSOC,
+            @ActionInput(name = "allow-ac-charging", label = "@text/actions.allow-ac-charging.label", description = "@text/actions.allow-ac-charging.description") boolean allowAcCharging,
+            @ActionInput(name = "start-time", label = "@text/actions.start-time.label", description = "@text/actions.start-time.description") String startTime,
+            @ActionInput(name = "stop-time", label = "@text/actions.stop-time.label", description = "@text/actions.stop-time.description") String stopTime,
+            @ActionInput(name = "program-enable", label = "@text/actions.program-enable.label", description = "@text/actions.program-enable.description") boolean programEnable) {
+        //
```

Comment:
What is the purpose of this comment?

Reply:
As documented https://www.openhab.org/docs/developer/bindings/#actions-bound-to-a-thing

Reply:
Just to make it look pretty on screen. ;)